### PR TITLE
Adds a flipping skillchip, the Chipped quirk, a chip connector implant, and a quirk to spawn with it.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Added the FOSBURY skillchip, found in hacked toy vendors.
This skillchip allows you to bypass the emote cooldown when flipping or spinning. However, if used too often, it will lower the chip's integrity. At first, this is harmless, emitting sparks only, but at a certain point, your head starts smoking, your brain begins short-circuiting, and once the chip has lost all integrity your head will explode in a shower of gore, giving you a cranial fissure!

Added the Chipped quirk. It allows you to spawn with one skillchip. Of note here is that the only chips you can spawn with are the 'default' skillchips found in the toy vendor, you can NOT spawn in with the fosbury chip or with the musical one.

Added the Chip Connector quirk. It contains a new implant, the chip connector, which allows you to install and take out skillchips at any time. Being EMPed will cause it to drop one random skillchip, and if you try using it while it's malfunctioning you'll take out a bit of your brain instead. It can also be made at the exosuit fabricator after basic research.

Renamed the organ slots for brain implants and made the connector implant take the CNS slot.
## Why It's Good For The Game

> Added the FOSBURY skillchip, found in hacked toy vendors.
This skillchip allows you to bypass the emote cooldown when flipping or spinning. However, if used too often, it will lower the chip's integrity. At first, this is harmless, emitting sparks only, but at a certain point, your head starts smoking, your brain begins short-circuiting, and once the chip has lost all integrity your head will explode in a shower of gore, giving you a cranial fissure!

During the time flipping was bugged to have no cooldown or delay, it was really funny seeing people flip at increasingly higher speeds. Many people miss it, and to be honest, so do I. But everything needs limits or it stops being funny and ends up being overdone. To this end this skillchip lets people relive those days of glory, until they go out in a beautiful explosion. The chip will last a very long while if well taken care of, and there are plenty of warning signs before it gets too risky to use, but we know what players are actually gonna do, and I am looking forward to it.

> Added the Chipped quirk. It allows you to spawn with one skillchip. Of note here is that the only chips you can spawn with are the 'default' skillchips found in the toy vendor, you can NOT spawn in with the fosbury chip or with the musical one.

Chips can be kind of neat sometimes, and I think always having one as part of your character can make for some fun things as the consistency of it makes it more likely to stick in your mind to do stuff with. 
> 'but can't you just go to the vendor at the start of every shift?'
Sure. But we have loadouts. We have tagger, musician, and many other things that are 'roundstart bloat', and while that by itself is not an excuse, it's part of the design of character setup: Minimizing the time spent running around the station for things your character is supposed to have for their personality/roleplay/gimmick. They're paying the cost of quirk points for it, anyways.

> Added the Chip Connector quirk. It contains a new implant, the chip connector, which allows you to install and take out skillchips at any time. Being EMPed will cause it to drop one random skillchip, and if you try using it while it's malfunctioning you'll take out a bit of your brain instead. It can also be made at the exosuit fabricator after basic research.

I like the idea of having a gross usb drive on the back of your brain that you shove chips into and out of, especially if it drops pieces of your brain when malfunctioning. It also adds a little bit of extra relevance to skillchips now that you don't need to spend 15 (!!!) seconds inside a skillstation to add one. The cost is, as stated, quite literally losing your brain on EMP's and the moderately expensive sum of 4 points.

> Renamed the organ slots for brain implants and made the connector implant take the CNS slot.

It annoys me that there were organ slots for 'anti drop implant'. Now they're just slots for anything meant to connect to the cerebellum/central nervous system, making it less brute forcey and also adding the start of some fun exclusivity between brain implants.
## Changelog
:cl:
add: Added the FOSBURY skillchip, found in hacked toy vendors. This skillchip allows you to bypass the emote cooldown when flipping or spinning. However, if used too often, it will lower the chip's integrity and cause malfunctions.
add: Added the Chipped quirk. It allows you to spawn with one base skillchip. (not the one above)
add: Added the Chip Connector quirk. It contains a new implant, the chip connector, which allows you to install and take out skillchips at any time.
code: Renamed the organ slots for brain implants and made the connector implant take the CNS slot.

### CODE REVIEW

**Comment by carlarctg** [comment]
I was considering making it so that the quirk chip connector is an older model that can't interface with some skillchips like the job ones, to limit its impact, but I'm not sure that's necessary.

**Comment by Xkallubar** [comment]
The chip connector has nothing to do with robotics, it should be printed at the sci lathe.

**Comment by necromanceranne** [comment]
> The chip connector has nothing to do with robotics, it should be printed at the sci lathe.

It's an implant. And also, robotics is a science department?

**Comment by Xkallubar** [comment]
> > The chip connector has nothing to do with robotics, it should be printed at the sci lathe.
> 
> It's an implant. And also, robotics is a science job?

Oh I thought it was a skillchip nevermind

File: code/datums/quirks/positive_quirks/chipped.dm [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		preferred_chip = GLOB.quirk_chipped_choice[pick(GLOB.quirk_chipped_choice)]
+
+	var/mob/living/carbon/quirk_holder_carbon = quirk_holder
+	if(iscarbon(quirk_holder))
+		installed_chip = new preferred_chip()
+		quirk_holder_carbon.implant_skillchip(installed_chip, force = TRUE)
+	installed_chip.try_activate_skillchip(silent = FALSE, force = TRUE)
+
+	var/obj/item/organ/internal/brain/itchy_brain = quirk_holder.get_organ_by_type(ORGAN_SLOT_BRAIN)
+	itchy_timer = addtimer(CALLBACK(itchy_brain, PROC_REF(cause_itchy), itchy_brain), rand(5 SECONDS, 10 MINUTES)) // they get The Itch from a poor quality install every so often
```
Comment (SmArtKar) [inline code/datums/quirks/positive_quirks/chipped.dm:46]: This won't work, callback should be on src but you're calling it on itchy_brain which does not have the cause_itchy proc

**Comment by necromanceranne** [comment]
I think an implant that interacts with skillchip swapping is a neat concept, though I do often ask myself how often I'd actually foresee myself either going to get multiple different chips, and how often I would be sitting on the same chips most of the round. Inserting skilllchips is relatively harmless as it is and fairly easy. You don't even need someone else to help you.

I then have to weigh my options; do I just go get the chips I want and not worry about swapping, or do I take the risk of this implant permanently fucking up my brain if things go badly with it to be able to swap in and out as I need. There are so few instances where I would need to get additional chips that I feel confident this isn't something I'd pursue until some more attractive options present themselves, and even then, I might still just go the standard, safe way?

Basically, I think has some cool potential and the downsides could be kind of funny to encounter, but the actual utilitarian benefits are a little mixed. 

One possible upside I could suggest is that it does actually increase you chip capacity, similar to the genetics mutation literally nobody gets. If you can slot more chips that normal, thats a decent upside (assuming you have so many in there to swap around too in the first place)

**Comment by carlarctg** [comment]
It's not meant to be a great implant, just a more immersive one for your oc with some side benefits and drawbacks. But I can increase complexity, sure

File: code/_globalvars/lists/quirks.dm [hunk 2]
```diff
@@ -98,3 +98,18 @@ GLOBAL_LIST_INIT(paraplegic_choice, list(
...
+	"WINE" = /obj/item/skillchip/wine_taster,
+	"Hedge 3" = /obj/item/skillchip/bonsai,
+	"Skillchip adapter" = /obj/item/skillchip/useless_adapter,
+	"N16H7M4R3" = /obj/item/skillchip/light_remover,
+	"3NTR41LS" = /obj/item/skillchip/entrails_reader,
+	"GENUINE ID Appraisal Now!" = /obj/item/skillchip/appraiser,
+	"Le S48R4G3" = /obj/item/skillchip/sabrage,
+	"Integrated Intuitive Thinking and Judging" = /obj/item/skillchip/intj,
+	"F0RC3 4DD1CT10N" = /obj/item/skillchip/drunken_brawler,
+	//"\"Space Station 13: The Musical\"" = /obj/item/skillchip/musical,
```
Comment (Ghommie) [inline code/_globalvars/lists/quirks.dm:114]: I think it'd be nice to have this enabled, but that's just my personal opinion.

Also for a moment I was like "woo, fishing quirk" when you said that skillchips from (unhacked) toy vending machines could be chosen, but I get why you didn't include the fishing skillchip. It's generally quite more useful than anything on this list.

Comment (carlarctg) [reply code/_globalvars/lists/quirks.dm:114]: Actually this pr was made before that. i'll go and enable'em

**Comment by EricZilla** [comment]
This thing is cool, adds to the options for starting as a low tech cyborg, love that

**Comment by YuliceTheDruid** [comment]
Make a deluxe version that has no downsides via cargo or for admins or something

**Comment by carlarctg** [comment]
...no just get emp protection

**Comment by optimumtact** [comment]
mixing the quirka nd sthe skillchip in one PR was such a bad idea.

**Comment by Autisem** [comment]
Why not just make new organ slot? feels wierd to choose between this and rebooter implant

File: code/game/objects/effects/spawners/gibspawner.dm [hunk 3]
```diff
@@ -71,6 +71,18 @@
...
+	gibtypes = list(/obj/effect/decal/cleanable/blood/gibs, /obj/effect/decal/cleanable/blood/gibs)
+	gibamounts = list(2, 2)
+	sound_vol = 40
+
+/obj/effect/gibspawner/human/smol/Initialize(mapload)
+	if(!gibdirections.len)
+		gibdirections = list(\
+			list(pick(GLOB.alldirs)),\
+			list(pick(GLOB.alldirs)),\
+			)
```
Comment (Ghommie) [inline code/game/objects/effects/spawners/gibspawner.dm:84]:
```suggestion
/obj/effect/gibspawner/generic/smol/Initialize(mapload)

	if(!length(gibdirections))
		gibdirections = list(
			list(pick(GLOB.alldirs)),
			list(pick(GLOB.alldirs)),
		)
```

File: code/modules/library/skill_learning/skillchip.dm [hunk 4]
```diff
@@ -602,3 +602,174 @@
...
+		It was pulled off the markets shortly after release due to users damaging the chip's integrity from excessive acrobatics, causing deadly malfunctions. It really puts the 'flop' in 'Fosbury Flop'!"
+	skill_name = "Spinesthetics"
+	skill_description = "Allows you to flip and spin at an illegal and dangerous rate."
+	skill_icon = FA_ICON_WHEELCHAIR_ALT
+	activate_message = span_notice("You suddenly have an extremely advanced and complex sense of how to spin and flip with grace.")
+	deactivate_message = span_notice("Your divine grasp of Spinesthesics disappears entirely.")
+	custom_premium_price = PAYCHECK_CREW * 4
+	// set integrity to 1 when mapping for !!FUN!!
+	max_integrity = 100
+	// more fun
```
Comment (Ghommie) [inline code/modules/library/skill_learning/skillchip.dm:618]: That's nary a bit descriptive, also use `///` for doc comments.

File: code/modules/library/skill_learning/skillchip.dm [hunk 5]
```diff
@@ -602,3 +602,174 @@
...
+				if(bozo.has_status_effect(/datum/status_effect/grouped/blindness))
+					to_chat(splashed, span_userdanger("You're covered in blood!"))
+				else
+					to_chat(splashed, span_userdanger("You are blinded by a shower of blood!"))
+				splashed.Stun(2 SECONDS)
+				splashed.set_eye_blur_if_lower(40 SECONDS)
+				splashed.adjust_confusion(3 SECONDS)
+
+			// GORE
+			/*
```
Comment (Ghommie) [inline code/modules/library/skill_learning/skillchip.dm:674]: If you aren't going to include it, remove this commented section.

File: code/modules/surgery/organs/internal/cyberimp/augments_internal.dm [hunk 6]
```diff
@@ -179,6 +179,107 @@
...
+	organ_flags |= ORGAN_FAILING
+	var/loops = 1
+	if(severity != EMP_LIGHT)
+		loops = 2
+	for(var/i in 1 to loops)
+		// you either lose a chip or a bit of your brain
+		owner.visible_message(span_warning("Something falls to the ground from behind [owner]'s head."),\
+			span_boldwarning("You feel something fall off from behind your head."))
+		var/obj/item/organ/internal/brain/chippy_brain = owner.get_organ_by_type(ORGAN_SLOT_BRAIN)
+		var/obj/item/skillchip/skillchip = chippy_brain?.skillchips[1]
```
Comment (Ghommie) [inline code/modules/surgery/organs/internal/cyberimp/augments_internal.dm:257]:
```suggestion
		var/obj/item/skillchip/skillchip = LAZYFIND(chippy_brain?.skillchips, 1)
```

File: code/modules/surgery/organs/internal/cyberimp/augments_internal.dm [hunk 7]
```diff
@@ -179,6 +179,107 @@
...
+		if(skillchip)
+			owner.remove_skillchip(skillchip, silent = TRUE)
+			skillchip.forceMove(owner.drop_location())
+			playsound(owner, 'sound/machines/terminal/terminal_eject.ogg', 25, TRUE)
+		else
+			remove_brain()
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 90 / severity)
```
Comment (Ghommie) [inline code/modules/surgery/organs/internal/cyberimp/augments_internal.dm:264]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(reboot)), 9 SECONDS / severity)
```

File: code/modules/surgery/organs/internal/cyberimp/augments_internal.dm [hunk 8]
```diff
@@ -179,6 +179,107 @@
...
+	to_chat(owner, span_warning("Your brain is empty!")) // heh
+
+/obj/item/organ/internal/cyberimp/brain/connector/emp_act(severity)
+	. = ..()
+	if((organ_flags & ORGAN_FAILING) || . & EMP_PROTECT_SELF)
+		return
+	organ_flags |= ORGAN_FAILING
+	var/loops = 1
+	if(severity != EMP_LIGHT)
+		loops = 2
```
Comment (Ghommie) [inline code/modules/surgery/organs/internal/cyberimp/augments_internal.dm:251]:
```suggestion
	var/loops = severity == EMP_LIGHT ? 1 : 2
```

File: code/modules/library/skill_learning/skillchip.dm [hunk 9]
```diff
@@ -602,3 +602,174 @@
...
+
+			sparks.set_up(rand(1,2), cardinals_only = TRUE, location = get_turf(src))
+			sparks.start()
+
+	return COMPONENT_EMOTE_COOLDOWN_BYPASS
+
+/obj/item/skillchip/acrobatics/proc/charge()
+	allowed_usage++
+
+/obj/item/skillchip/acrobatics/kiss
```
Comment (Ghommie) [inline code/modules/library/skill_learning/skillchip.dm:763]: This isn't used anywhere. I think you can easily remove this mistake.

File: code/modules/library/skill_learning/skillchip.dm [hunk 10]
```diff
@@ -602,3 +602,174 @@
...
+	switch(rand(mintegrity, get_integrity())) // 1 to 100 but gets worse every time
+		// CRIT FAIL
+		if(1)
+			bozo.visible_message(span_userdanger("[bozo]'s head suddenly explodes outwards!"))
+
+			explosion(bozo, light_impact_range = 2, adminlog = TRUE, explosion_cause = src)
+			// WITNESS THE GORE
+			for(var/mob/living/splashed in view(2, bozo))
+				if(bozo.has_status_effect(/datum/status_effect/grouped/blindness))
+					to_chat(splashed, span_userdanger("You're covered in blood!"))
```
Comment (Ghommie) [inline code/modules/library/skill_learning/skillchip.dm:666]:
```suggestion
					to_chat(splashed, span_userdanger("You're splashed with something!"))
```
If you're blind, how can you tell it's exactly blood (except from the slight smell of iron)?

File: code/modules/library/skill_learning/skillchip.dm [hunk 11]
```diff
@@ -602,3 +602,174 @@
...
+				else
+					to_chat(splashed, span_userdanger("You are blinded by a shower of blood!"))
+				splashed.Stun(2 SECONDS)
+				splashed.set_eye_blur_if_lower(40 SECONDS)
```
Comment (Ghommie) [inline code/modules/library/skill_learning/skillchip.dm:670]: That's way too high imo.

File: tgstation.dme [hunk 12]
```diff
@@ -1794,6 +1794,8 @@
 #include "code\datums\quirks\positive_quirks\alcohol_tolerance.dm"
 #include "code\datums\quirks\positive_quirks\apathetic.dm"
 #include "code\datums\quirks\positive_quirks\bilingual.dm"
+#include "code\datums\quirks\positive_quirks\chipped.dm"
+#include "code\datums\quirks\positive_quirks\chip_connector.dm"
```
Comment (Ghommie) [inline tgstation.dme:1798]:
```suggestion
#include "code\datums\quirks\positive_quirks\chip_connector.dm"
#include "code\datums\quirks\positive_quirks\chipped.dm"
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/quirks/positive_quirks/chipped.dm (hunk 1 above)
Comment (carlarctg) [reply code/datums/quirks/positive_quirks/chipped.dm:46]: why didnt it rutnime

File: code/modules/library/skill_learning/skillchip.dm (hunk 10 above)
Comment (carlarctg) [reply code/modules/library/skill_learning/skillchip.dm:666]: anosmia content

File: code/modules/library/skill_learning/skillchip.dm (hunk 5 above)
Comment (carlarctg) [reply code/modules/library/skill_learning/skillchip.dm:674]: i couldnt get it to drop an organ and its driving me mad..........

File: code/modules/library/skill_learning/skillchip.dm (hunk 9 above)
Comment (carlarctg) [reply code/modules/library/skill_learning/skillchip.dm:763]: is there harm in keeping it as an example for mappers and coders. its funny

Comment (Ghommie) [reply code/modules/library/skill_learning/skillchip.dm:763]: _shrugs_

File: code/game/objects/effects/spawners/gibspawner.dm (hunk 3 above)
Comment (carlarctg) [reply code/game/objects/effects/spawners/gibspawner.dm:84]: fucking hate this shitty effect

File: tgstation.dme (hunk 12 above)
Comment (Ghommie) [reply tgstation.dme:1798]: This is still an issue, man.

**Comment by uaioy** [comment]
free funny thing to annoy people with and free reason to exit round when non antag whats there to not love bros

**Review by Ghommie — APPROVED** [review]
